### PR TITLE
Facilities to create actions from lambdas

### PR DESCRIPTION
**Pull request by atrantan**
This PR tries to give some facilities to create actions from lambdas. It is inspired from the trick explained in http://pfultz2.com/blog/2014/09/02/static-lambda/. The constraints for the lambda are :

- No captured variables (property checked via `static_assert()`)
- Parameters passed by value (similarly to global functions that are converted into actions)

### CODE REVIEW

File: tests/regressions/actions/plain_action_move_semantics.cpp [hunk 1]
```diff
@@ -45,6 +45,17 @@ std::size_t pass_movable_object(movable_object const& obj)
     return obj.get_count();
 }
 
+constexpr auto movable_object_void_passer =
```
Comment (hkaiser) [inline tests/regressions/actions/plain_action_move_semantics.cpp]: Please use `HPX_CONSTEXPR` instead of the plain `constexpr`.

Comment (atrantan) [inline tests/regressions/actions/plain_action_move_semantics.cpp]: Done in commit 3372a4f582ca634f9ea923b5dee7a81ba742353c

**Comment by hkaiser** [comment]
@atrantan This looks good to me now. One minor caveat: I think all of this will work with C++14 enabled only. Would it be possible to surround the code with `#ifdef HPX_HAVE_CXX14` and to issue an `#error` if it is being used otherwise?

**Comment by atrantan** [comment]
@hkaiser I suggest to try to compile the code with C++11 before. For the moment It passed with gcc when I use `-std=c++11`

**Comment by hkaiser** [comment]
@atrantan: Ahh, very good. Let's merge this, then.

**Pull request merged**